cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -365,35 +398,34 @@ pub mod tests {
         // Do the build to ensure it's been run.  We just want to find the outputs
         assert_build_status(&flox, &mut env, EXAMPLE_PACKAGE_NAME, true);
 
-        let meta = check_build_metadata(&env, EXAMPLE_PACKAGE_NAME).unwrap();
+        let meta = check_build_metadata(&env, EXAMPLE_PACKAGE_NAME, &flox.system).unwrap();
         assert_eq!(meta.outputs.len(), 1);
         assert_eq!(meta.outputs[0].store_path.starts_with("/nix/store/"), true);
     }
 
-    // Template end to end test
-    // #[test]
-    // fn test_create_provider() {
-    //     let (flox, _temp_dir_handle) = flox_instance();
-    //     let (_tempdir_handle, _remote_repo, remote_uri) = example_remote();
-    //     let (env, build_repo) = example_path_environment(&flox, Some(&remote_uri));
-
-    //     let publish_provider = PublishProvider {
-    //         _base_temp_dir: PathBuf::from("/tmp"),
-    //         _auth_token: None,
-    //         _builder: &example_builder(flox, ),
-    //     };
-
-    //         // todo - test floxhub_token -> catalog name logic
-
-    //     let (_tempdir_handle, _remote_repo, remote_uri) = example_remote();
-    //     let (env, _git) = example_path_environment(&flox, Some(&remote_uri));
-    //     let _res = publish_provider.publish(
-    //         &flox,
-    //         Client::Mock(Default::default()),
-    //         PublishEnvironment::Path(env),
-    //         example_builder(),
-    //         EXAMPLE_PACKAGE_NAME,
-    //     );
-
-    // }
+    #[test]
+    fn test_publish() {
+        let (flox, _temp_dir_handle) = flox_instance();
+        let (_tempdir_handle, _remote_repo, remote_uri) = example_remote();
+        let (mut env, _build_repo) = example_path_environment(&flox, Some(&remote_uri));
+
+        // Do the build to ensure it's been run.  We just want to find the outputs
+        assert_build_status(&flox, &mut env, EXAMPLE_PACKAGE_NAME, true);
+
+        let client = Client::Mock(MockClient::new(None::<String>).unwrap());
+        let token = create_test_token("test").unwrap();
+
+        let (env_meta, build_meta) = (
+            check_environment_metadata(&flox, &env).unwrap(),
+            check_build_metadata(&env, EXAMPLE_PACKAGE_NAME, &flox.system).unwrap(),
+        );
+
+        let publish_provider = PublishProvider::<&FloxBuildMk> {
+            build_meta,
+            env_meta,
+            _builder: None,
+        };
+
+        let _res = publish_provider.publish(&client, &token);
```

Comment:
suggestion followup: I think to actually test this we should make an assertion about what `publish_build` gets called with, which would require some changes to the mock client.

I think for the catalog it was problematic a few times that we did mocking at the catalog provider level rather than at the `catalog_api_v1` level. Not sure what the testing plan is for publish and if `providers::catalog` is just a passthrough it's probably easiest to follow the same pattern, but if there's ever going to be any real logic in `providers::catalog` we should probably mock one level deeper

Reply:
Noted for follow up. Thanks.